### DISCUSSION TITLE
`geometry-type` spec implementation misalignment

### DISCUSSION THREAD

**Discussion by HarelM** [maplibre/maplibre-style-spec#536] [Polls]
As was discussed here:
- https://github.com/maplibre/maplibre-gl-js/issues/3516

There is a misalignment between the spec and the implementation of `geometry-type` which we would like to resolve.
The problem is that `geometry-type` works much like `$type` and does not return "multi" type of features, which is basically a duplicated operator.
In order to resolve it we would like to get more feedback from the community.
Below are the two options:

**Comment by wipfli** [comment]
Oh I clicked the wrong button and can't change it now.

I think we should keep the current behavior and update the docs. The style spec is not perfect, it has bugs, and there are people who rely on these bugs. It is an old library after all.

I think people rely more on what the renderer does than what is written somewhere in the docs.

**Reply by louwers** [reply]
I agree, but if this change is introduced in a major (breaking) version, it is better than introducing yet another operator for those that need this functionality.

**Reply by wipfli** [reply]
We do not have versioning for the style spec, despite the fact that the style spec package has a version number.

**Reply by wipfli** [reply]
So for probably like 10 years there has not been a breaking change I think in the style spec

**Reply by wipfli** [reply]
OK that is actually not quite correct. The style specification has a version and it is pinned to 8, see https://maplibre.org/maplibre-style-spec/root/#version.
